docs/fundamentals/code-analysis/quality-rules/ca2255.md

```diff
@@ -0,0 +1,40 @@
+---
+title: "CA2255: The `ModuleInitializer` attribute should not be used in libraries"
+description: "Learn about code analysis rule CA2255: The `ModuleInitializer` attribute should not be used in libraries"
+ms.date: 11/15/2021
+ms.topic: reference
+f1_keywords:
+- CA2255
+- ModuleInitializerAttributeShouldNotBeUsedInLibraries
+helpviewer_keywords:
+- ModuleInitializerAttributeShouldNotBeUsedInLibraries
+- CA2255
+author: jeffhandley
+---
+# CA2255: The `ModuleInitializer` attribute should not be used in libraries
+
+| | Value |
+|-|-|
+| **Rule ID** |CA2255|
+| **Category** |[Usage](usage-warnings.md)|
+| **Fix is breaking or non-breaking** |Breaking|
+
+## Cause
+
+Applying <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute> to a method within a Class Library.
+
+## Rule description
+
+Module initializers are intended to be used by application code to ensure an application's components are initialized before the application code begins executing. If library code declares a <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute> method, it can interfere with application initialization and also lead to limitations in that application's trimming abilities. Library code should therefore not utilize the <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute> attribute.
+
+## How to fix violations
+
+Instead of using <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute> methods, the library should expose methods that can be used to initialize any components within the library, and allow the application to invoke the method during application initialization.
```

Comment:
```suggestion
Instead of using methods with the <xref:System.Runtime.CompilerServices.ModuleInitializerAttribute>, a library should expose methods that can be used to initialize any components within the library, and allow the application to invoke the method during application initialization.
```